DMX.Core.Api.Tests.Unit/Services/Foundations/Labs/LabServiceTests.Exceptions.Remove.cs

```diff
@@ -0,0 +1,174 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using DMX.Core.Api.Models.Foundations.Labs;
+using DMX.Core.Api.Models.Foundations.Labs.Exceptions;
+using FluentAssertions;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Foundations.Labs
+{
+    public partial class LabServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyValidationOnRemoveIfDatabaseUpdateConcurrencyErrorOccursAndLogItAsync()
+        {
+            // given
+            var someGuid = Guid.NewGuid();
+            Lab someLab = CreateRandomLab();
+
+            var dbUpdateConcurrencyException =
+                new DbUpdateConcurrencyException();
+
+            var lockedLabException =
+                new LockedLabException(dbUpdateConcurrencyException);
+
+            var expectedLabDependencyValidationException =
+                new LabDependencyValidationException(lockedLabException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectLabByIdAsync(It.IsAny<Guid>()))
+                    .ReturnsAsync(someLab);
```

Comment:
Why don't you throw the exception from here instead?

Reply:
Same thing for the rest of these tests.

Reply:
We like a challenge :)